From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 28 Feb 2020 16:56:29 +0100
Subject: filechooser: Fix crash when file has no content-type

This might happen for slow filesystems where a fast-content-type might
be provided instead. Don't try to manipulate that content_type if it's
NULL, otherwise we'll either throw warnings (at best) or crash (at
worse).

Origin: upstream, 3.24.15
---
 gtk/gtkfilechooserwidget.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gtk/gtkfilechooserwidget.c b/gtk/gtkfilechooserwidget.c
index 9c7aadb..2e98325 100644
--- a/gtk/gtkfilechooserwidget.c
+++ b/gtk/gtkfilechooserwidget.c
@@ -5049,6 +5049,9 @@ get_type_information (GtkFileChooserWidget *impl,
   char *description;
 
   content_type = g_file_info_get_content_type (info);
+  if (!content_type)
+    goto end;
+
   switch (impl->priv->type_format)
     {
     case TYPE_FORMAT_MIME:
@@ -5066,6 +5069,7 @@ get_type_information (GtkFileChooserWidget *impl,
       g_assert_not_reached ();
     }
 
+end:
   return g_strdup ("");
 }
 
